sdk/communication/AzureCommunication/Tests/ObjCCommunicationUserCredentialAsyncTests.m

```diff
@@ -0,0 +1,159 @@
+                                             NSError * _Nullable error) {
+        XCTAssertNotNil(accessToken);
+        XCTAssertNil(error);
+        XCTAssertEqual(accessToken.token, weakSelf.sampleToken);
+        XCTAssertEqual(accessToken.expiresOn.timeIntervalSince1970, weakSelf.sampleTokenExpiry);
+        
+        [expectation fulfill];
+    }];
+    
+    [self waitForExpectations:@[expectation] timeout:5.0];
+}
+
+- (void)test_ObjCRefreshTokenProactively_TokenExpiringInOneMin {
+    XCTestExpectation *expectation = [self expectationWithDescription:
+                                      @"RefreshTokenProactively_TokenExpiringInOneMin"];
+    __weak ObjCCommunicationUserCredentialAsyncTests *weakSelf = self;
+    
+    NSString *token = [self generateTokenValidForMinutes: 1];
+    CommunicationUserCredential *credential = [[CommunicationUserCredential alloc]
+                                               initWithInitialToken:token
+                                               refreshProactively:YES
+                                               error:nil
+                                               tokenRefresher:
+                                               ^(void (^ block)
+                                                 (NSString * _Nullable newToken,
+                                                  NSError * _Nullable error)) {
+        weakSelf.fetchTokenCallCount += 1;
+        block(weakSelf.sampleToken, nil);
+    }];
+    
+    [credential tokenWithCompletionHandler:^(CommunicationAccessToken * _Nullable accessToken,
+                                             NSError * _Nullable error) {
+        XCTAssertNotNil(accessToken);
+        XCTAssertEqual(accessToken.token, weakSelf.sampleToken);
+        XCTAssertEqual(weakSelf.fetchTokenCallCount, 1);
+        
+        [expectation fulfill];
+    }];
+
+    [self waitForExpectations:@[expectation] timeout:2.0];
+}
+
+- (void)test_ObjCRefreshTokenProactively_TokenExpiringInNineMin {
+    XCTestExpectation *expectation = [self expectationWithDescription:
+                                      @"RefreshTokenProactively_TokenExpiringInNineMin"];
+    __weak ObjCCommunicationUserCredentialAsyncTests *weakSelf = self;
+    
+    NSString *token = [self generateTokenValidForMinutes: 9];
+    CommunicationUserCredential *credential = [[CommunicationUserCredential alloc]
+                                               initWithInitialToken:token
+                                               refreshProactively:YES
+                                               error:nil
+                                               tokenRefresher:
+                                               ^(void (^ block)
+                                                 (NSString * _Nullable newToken,
+                                                  NSError * _Nullable error)) {
+        weakSelf.fetchTokenCallCount += 1;
```

Comment:
Do we have assert to validate the value of weakSelf.fetchTokenCallCount ?

Reply:
Yes this is being asserted in line 131